base/reduce.jl

```diff
@@ -1109,3 +1309,8 @@ for (fred, f) in ((maximum, max), (minimum, min), (sum, add_sum))
         isempty(r) ? init : op(init, $fred(r))
     end
 end
+
+# Reduction kernels that explicitly encode simplistic SIMD-ish reorderings
+const CommutativeOps = Union{typeof(+),typeof(Base.add_sum),typeof(min),typeof(max),typeof(Base._extrema_rf),typeof(|),typeof(&)}
+is_commutative_op(::CommutativeOps) = true
```

Comment:
maybe two arg is better with a default?
```
is_commutative_op(op, ::Type) = is_commutative_op(op)
```

so we can say like `*` on `Float64` commutes (for the purposes of `reduce`) but not on `Matrix` or quaternions or whatever